crates/ruff_python_parser/src/lexer.rs

```diff
@@ -2051,6 +2059,22 @@ def f(arg=%timeit a = b):
         assert_debug_snapshot!(lex_source(source));
     }
 
+    #[test]
+    fn test_fstring_with_multiline_format_spec() {
+        let source = r"f'''__{
+    x:d
+}__'''
+f'''__{
+    x:a
+        b
+          c
+}__'''
```

Comment:
How should this one be interpreted? What is the intended "meaning"? Does this have a format spec, or no?

Reply:
It does have the format spec and it contains newlines. The interpretation of the actual value is left to the runtime but the parser should preserve the newlines. The reason this is the case for triple-quoted f-strings is to maintain backwards compatibility as the following code works on 3.11 and earlier as well:

```
In [3]: f"""{datetime.datetime.now():%Y
   ...:     %m
   ...:     %d}"""
Out[3]: '2023\n    10\n    05'
```